### ISSUE TITLE
Easy way to change the file system type?

### ISSUE THREAD

**Issue by astrophys** [xcat2/xcat-core#7266]
We are using xcat-2.16.3 and our osimages are RHEL-8.4.  We want the root filesystem to be ext4 instead of xfs, so kdump can write kernel core files to the local filesystem. Filesystem type (FSTYPE) looks to be fixed in pre.rhel8 script. As far as we can tell, the only mechanism to change FSTYPE is to set a custom partition script, which requires that we recreate /tmp/partitionfile in whole. Is there a simpler way to change FSTYPE for the root filesystem?

**Comment by besawn** [comment]
> We want the root filesystem to be ext4 instead of xfs, so kdump can write kernel core files to the local filesystem.

Local xfs file systems are a supported kdump target according to the official RHEL 8 documentation: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/managing_monitoring_and_updating_the_kernel/supported-kdump-configurations-and-targets_managing-monitoring-and-updating-the-kernel#supported-kdump-targets_supported-kdump-configurations-and-targets

Can you elaborate on why you think you need to change the root file system to ext4 to enable kdump?

> Is there a simpler way to change FSTYPE for the root filesystem?

Creating a custom partition file is the the simplest way to change the root file system to ext4. 

There is some additional information here:
https://xcat-docs.readthedocs.io/en/stable/guides/admin-guides/manage_clusters/ppc64le/diskful/customize_image/cfg_partition.html

Here is an example partition file generated by xCAT when installing `rhels8.4.0-ppc64le-install-compute` on a test VM:
```
# cat /tmp/partitionfile 
ignoredisk --only-use=/dev/sda
part prepboot --fstype=prepboot --asprimary --ondisk=/dev/sda --size=8
part /boot --fstype=xfs --asprimary --ondisk=/dev/sda --size=1024
part pv.000997 --grow --asprimary --ondisk=/dev/sda --size=1024
volgroup xcatvg --pesize=4096 pv.000997
logvol swap --name=swap --vgname=xcatvg --recommended
logvol / --fstype=xfs --name=root --vgname=xcatvg --size=1024 --grow 
bootloader --boot-drive=sda
```
You can create a partition file using the above example as a starting point, using the kickstart documentation as a starting point, or you can see exactly what xCAT is automatically generating as the partition file for your node by doing the following:
Enable `xcatdebugmode=2` in the site table, use `tabedit site` and modify or add the following line:
```
"xcatdebugmode","2",,
```
```
# tabedit site
# tabdump site | grep xcatdebugmode
"xcatdebugmode","2",,
```

this will allow you to ssh into the node while the diskful install is happening and look at the contents of `/tmp/partitionfile`.
To do that, start a fresh install of the node:
`[root@mn ~]# rinstall c910f03c09k17 osimage=rhels8.4.0-ppc64le-install-compute`

Wait for the install to start, then ssh into the node to check the file contents:
```
[root@mn ~]# ssh c910f03c09k17
[anaconda root@c910f03c09k17 ~]# cat /tmp/partitionfile
ignoredisk --only-use=/dev/sda
part prepboot --fstype=prepboot --asprimary --ondisk=/dev/sda --size=8
part /boot --fstype=xfs --asprimary --ondisk=/dev/sda --size=1024
part pv.000997 --grow --asprimary --ondisk=/dev/sda --size=1024
volgroup xcatvg --pesize=4096 pv.000997
logvol swap --name=swap --vgname=xcatvg --recommended
logvol / --fstype=xfs --name=root --vgname=xcatvg --size=1024 --grow 
bootloader --boot-drive=sda
```
Once you have decided what you want your partition file to look like, associate it with your osimage and re-install the node:
```
[root@mn ~]# vi /install/custom/my-partitions

[root@mn ~]# cat /install/custom/my-partitions
ignoredisk --only-use=/dev/sda
part prepboot --fstype=prepboot --asprimary --ondisk=/dev/sda --size=8
part /boot --fstype=xfs --asprimary --ondisk=/dev/sda --size=1024
part pv.000997 --grow --asprimary --ondisk=/dev/sda --size=1024
volgroup xcatvg --pesize=4096 pv.000997
logvol swap --name=swap --vgname=xcatvg --recommended
logvol / --fstype=ext4 --name=root --vgname=xcatvg --size=1024 --grow 
bootloader --boot-drive=sda

[root@mn ~]# chdef -t osimage rhels8.4.0-ppc64le-install-compute partitionfile=/install/custom/my-partitions
1 object definitions have been created or modified.

[root@mn ~]# lsdef -t osimage rhels8.4.0-ppc64le-install-compute -i partitionfile
Object name: rhels8.4.0-ppc64le-install-compute
    partitionfile=/install/custom/my-partitions

[root@mn ~]# rinstall c910f03c09k17 osimage=rhels8.4.0-ppc64le-install-compute
Provision node(s): c910f03c09k17
```
```
After installation completes:
[root@c910f03c09k17 ~]# lsblk
NAME            MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
sda               8:0    0   40G  0 disk 
├─sda1            8:1    0    8M  0 part 
├─sda2            8:2    0    1G  0 part /boot
└─sda3            8:3    0   39G  0 part 
  ├─xcatvg-root 253:0    0   35G  0 lvm  /
  └─xcatvg-swap 253:1    0    4G  0 lvm  [SWAP]
sr0              11:0    1 1024M  0 rom  

[root@c910f03c09k17 ~]# df -hlT
Filesystem              Type      Size  Used Avail Use% Mounted on
devtmpfs                devtmpfs  3.7G     0  3.7G   0% /dev
tmpfs                   tmpfs     3.8G     0  3.8G   0% /dev/shm
tmpfs                   tmpfs     3.8G   15M  3.8G   1% /run
tmpfs                   tmpfs     3.8G     0  3.8G   0% /sys/fs/cgroup
/dev/mapper/xcatvg-root ext4       35G  1.7G   31G   6% /
/dev/sda2               xfs      1014M  237M  778M  24% /boot
tmpfs                   tmpfs     764M     0  764M   0% /run/user/0
```

In the example above, I started with the xCAT generated partition file (with xfs root file system) and then modified it to change the root file system to ext4.

However, it's unclear to me why you need to make this change to enable kdump.

Also, change xcatdebugmode back to 0 in the site table when you are done experimenting:
```
# tabedit site
# tabdump site | grep xcatdebugmode
"xcatdebugmode","0",,
```

**Comment by astrophys** [comment]
Evidently we needed to properly configure our `/etc/kdump.conf` and specify the current file system.  We got the kernel dumps to work on our xfs file system.  The explanation regarding (the now unnecessary) file system partitioning does answer some questions about how the `partitionfile` works and also I was unaware of the `xcatdebugmode`, which will be very helpful.  Thank you for the detailed response.
